pkg/controllers/nodeclass/status/ami.go

```diff
@@ -58,9 +62,15 @@ func (a *AMI) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (reconc
 		return v1.AMI{
 			Name:         ami.Name,
 			ID:           ami.AmiID,
+			Deprecated:   lo.Ternary(ami.Deprecated, lo.ToPtr(ami.Deprecated), nil),
 			Requirements: reqs,
 		}
 	})
-	nodeClass.StatusConditions().SetTrue(v1.ConditionTypeAMIsReady)
+	// Set the status condition on the nodeclass if any discovered AMI for the selector terms are deprecated or not
+	if containsDeprecatedAMIs {
+		nodeClass.StatusConditions().SetTrue(v1.ConditionTypeAMIsDeprecated)
```

Comment:
Can we add this condition to our docs? https://karpenter.sh/docs/concepts/nodeclasses/#statusconditions

Reply:
Updated!